pennylane/transforms/decompositions/clifford_t/clifford_t_transform.py

```diff
@@ -480,7 +480,10 @@ def circuit(x, y):
                 decomp_ops.extend(clifford_ops)
             else:
                 decomp_ops.append(op)
-        decomp_ops.append(qml.GlobalPhase(phase))
+
+        # check if phase is non-zero for non jax-jit cases
+        if not qml.math.is_abstract(phase) and not qml.math.allclose(phase, 0.0):
```

Comment:
i think we should always add it if the phase is abstract, since it will often be non-zero
```suggestion
        if qml.math.is_abstract(phase) or not qml.math.allclose(phase, 0.0):
```